libc/__fprintf_chk.c

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) Open Enclave SDK contributors.
+// Licensed under the MIT License.
+
+#ifdef __GNUC__
+
+#include <openenclave/enclave.h>
+#include <stdarg.h>
+#include <stdio.h>
+
+// GCC sometimes replaces fprintf() calls with __fprintf_chk() calls. In glibc
```

Comment:
What version of GCC is this and in what situations are you seeing these calls generated? If code is compiled with `-nostdlib`, calls to code in glibc shouldn't be linked, right?

Reply:
I think it happened when adding a static library to my enclave. I guess this is not supposed to be supported and one should recompile such dependencies?

Reply:
I could go either way. OE provides a standard C library, so I don't know that dependencies should all need to be compiled with nostdlib (or some other combination of compiler arguments). But supporting every internal function for every compiler could get a bit out of hand. I guess if someone wanted to target a different libc implementation in a non-enclave application they would have to recompile the dependencies in the same way.

@anakrish @CodeMonkeyLeet do either of you have an opinion here?

Reply:
GCC emits calls to these object size checking functions when compiled with _FORTIFY_SOURCE and optimizations https://gcc.gnu.org/onlinedocs/gcc/Object-Size-Checking.html

Secure coding practices recommend compiling with _FORTIFY_SOURCE (and also other flags like -fstack-protector) https://wiki.debian.org/Hardening

See https://github.com/openenclave/openenclave/issues/1600 for a list of secure flags that OE must be built with.

Reply:
Ok thats fine. For reference, BSD libc just aliases the functions (i.e. forwards __fprintf_chk to fprintf).

So I'm fine with supporting this in OE.